# HG changeset patch
# User adf88@interia.pl
# Date 1482591957 -3600
#      Sat Dec 24 16:05:57 2016 +0100
# Branch trunk
# Node ID 2b29bfc5fcaaf89c169998c6a0d25a67017968ee
# Parent  101381b218b07e7e6b2dba3563e8c62026574c32
[mq]: 1.fix-clipping-by-window-clip.diff

diff -r 101381b218b0 -r 2b29bfc5fcaa src/window.cpp
--- a/src/window.cpp	Fri Dec 23 22:13:38 2016 +0000
+++ b/src/window.cpp	Sat Dec 24 16:05:57 2016 +0100
@@ -942,7 +942,7 @@
 				left < w->left + w->width &&
 				top < w->top + w->height) {
 			/* Window w intersects with the rectangle => needs repaint */
-			DrawOverlappedWindow(w, left, top, right, bottom);
+			DrawOverlappedWindow(w, max(left, w->left), max(top, w->top), min(right, w->left + w->width), min(bottom, w->top + w->height));
 		}
 	}
 }
